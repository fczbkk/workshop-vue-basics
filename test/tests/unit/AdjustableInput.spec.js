import { shallowMount } from '@vue/test-utils'
import AdjustableInput from './../../src/components/AdjustableInput'

describe('AdjustableInput', () => {

  it('should start at 0', () => {
    const wrapper = shallowMount(AdjustableInput)
    expect(wrapper.vm.inputValue).toEqual(0)
  })

  it('should use provided value', () => {
    const wrapper = shallowMount(AdjustableInput, {propsData: {value: 42}})
    expect(wrapper.vm.inputValue).toEqual(42)
  })

  it('should react to clicking on add button', () => {
    const wrapper = shallowMount(AdjustableInput)
    const button = wrapper.find({ref: 'addButton'})
    button.trigger('click')
    expect(wrapper.vm.inputValue).toEqual(1)
  })

  it('should not go below zero', () => {
    const wrapper = shallowMount(AdjustableInput, {propsData: {value: 0}})
    const button = wrapper.find({ref: 'removeButton'})
    button.trigger('click')
    expect(wrapper.vm.inputValue).toEqual(0)
  })

  it('should render correctly', function () {
    const wrapper = shallowMount(AdjustableInput)
    expect(wrapper.element).toMatchSnapshot()
  })

})
