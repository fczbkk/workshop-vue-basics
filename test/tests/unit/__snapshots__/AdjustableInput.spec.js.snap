// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdjustableInput should render correctly 1`] = `
<div>
  <button>
    -
  </button>
   
  <input />
   
  <button>
    +
  </button>
</div>
`;
